plugins/home/src/components/CustomHomepage/AddWidgetDialog.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Widget } from './types';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import { DialogContent, DialogTitle, ListItemAvatar } from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import ListItemText from '@material-ui/core/ListItemText';
+import React from 'react';
+
+interface AddWidgetDialogProps {
+  widgets: Widget[];
+  handleAdd: (widgetName: string) => void;
+}
+
+export const AddWidgetDialog = (props: AddWidgetDialogProps) => {
+  const { widgets, handleAdd } = props;
+  return (
+    <>
+      <DialogTitle>Add new widget to dashboard</DialogTitle>
+      <DialogContent>
+        <List dense>
+          {widgets.map((child, index) => {
+            return (
+              <ListItem
+                key={index}
```

Comment:
Example to comment bellow: Would be nice to use an Widget `id` here

Reply:
We could use name instead